astropy/utils/metadata.py

```diff
@@ -81,7 +176,18 @@ def merge(left, right, merge_func=concat, metadata_conflicts='warn'):
 
         else:
             try:
-                out[key] = merge_func(left[key], right[key])
+                if merge_func is None:
+                    for left_type, right_type, merge_cls in MERGE_STRATEGY_CLASSES:
+                        if not merge_cls.enabled:
```

Comment:
Ah, I like the way `add_enabled_units` works!   So the classes can be defined without the `enabled` attribute, and there is choice of then permanently adding one or more merge strategies, or doing something within a context manager.  This eliminates the need for a separate function to list enabled strategies, since this would simply the be module-level `MERGE_STRATEGY_CLASSES`.